tensorflow/contrib/seq2seq/python/ops/layers.py

```diff
@@ -20,15 +20,185 @@
+  from Sutskever et al., 2014 https://arxiv.org/abs/1409.3215, at the
+  beginning of decoding.
+
+  The parameter sequence length is nessesary as it determines how many
+  timesteps to decode for each sample. TODO: Could make it optional for
+  training.
+
+  Args:
+    cell: An instance of RNNCell.
+    inputs: The RNN inputs.
+      If `time_major == False` (default), this must be a `Tensor` of shape:
+        `[batch_size, max_time, ...]`.
+      If `time_major == True`, this must be a `Tensor` of shape:
+        `[max_time, batch_size, ...]`.
+      The input to `cell` at each time step will be a `Tensor` with dimensions
+      `[batch_size, ...]`.
+    sequence_length: An int32/int64 vector sized `[batch_size]`.
+    initial_state: An initial state for the RNN.
+      Must be [batch_size, num_features], where num_features does not have to
+      match the cell.state_size. As a projection is performed at the beginning
+      of the decoding.
+    decoder_fn: A function that takes a state and returns an embedding.
+      The decoder function is closely related to `_extract_argmax_and_embed`.
+      Here is an example of a `decoder_fn`:
+      def decoder_fn(embeddings, weight, bias):
+        def dec_fn(state):
+          prev = tf.matmul(state, weight) + bias
+          return tf.gather(embeddings, tf.argmax(prev, 1))
+        return dec_fn
+    encoder_projection: (optional) given that the encoder might have a
+      different size than the decoder, we project the intial state as
+      described in Bahdanau, 2014 (https://arxiv.org/abs/1409.0473).
+      The optional `encoder_projection` is a
+      `tf.contrib.layers.fully_connected` with
+      `activation_fn=tf.python.ops.nn.tanh`.
+    parallel_iterations: (Default: 32).  The number of iterations to run in
+      parallel.  Those operations which do not have any temporal dependency
+      and can be run in parallel, will be.  This parameter trades off
+      time for space.  Values >> 1 use more memory but take less time,
+      while smaller values use less memory but computations take longer.
+    swap_memory: Transparently swap the tensors produced in forward inference
+      but needed for back prop from GPU to CPU.  This allows training RNNs
+      which would typically not fit on a single GPU, with very minimal (or no)
+      performance penalty.
+    time_major: The shape format of the `inputs` and `outputs` Tensors.
+      If true, these `Tensors` must be shaped `[max_time, batch_size, depth]`.
+      If false, these `Tensors` must be shaped `[batch_size, max_time, depth]`.
+      Using `time_major = True` is a bit more efficient because it avoids
+      transposes at the beginning and end of the RNN calculation.  However,
+      most TensorFlow data is batch-major, so by default this function
+      accepts input and emits output in batch-major form.
+    scope: VariableScope for the created subgraph; defaults to "RNN".
+  Returns:
```

Comment:
this is completely inconsistent with your code; which returns something really different.